pkg/cmd/cli/cmd/rsync/rsync.go

```diff
@@ -229,7 +236,93 @@ func (o *RsyncOptions) Validate() error {
 
 // RunRsync copies files from source to destination
 func (o *RsyncOptions) RunRsync() error {
-	return o.Strategy.Copy(o.Source, o.Destination, o.Out, o.ErrOut)
+	if err := o.Strategy.Copy(o.Source, o.Destination, o.Out, o.ErrOut); err != nil {
+		return err
+	}
+
+	if !o.Watch {
+		return nil
+	}
+	return o.WatchAndSync()
+}
+
+// WatchAndSync sets up a recursive filesystem watch on the sync path
+// and invokes rsync each time the path changes.
+func (o *RsyncOptions) WatchAndSync() error {
+
+	// these variables must be accessed while holding the changeLock
+	// mutex as they are shared between goroutines to communicate
+	// sync state/events.
+	var (
+		changeLock sync.Mutex
+		dirty      bool
+		lastChange time.Time
+		watchError error
+	)
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf("error setting up filesystem watcher: %v", err)
+	}
+	defer watcher.Close()
+
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				changeLock.Lock()
+				glog.V(5).Infof("filesystem watch event: %s", event)
+				lastChange = time.Now()
+				dirty = true
+				// Rename sends a rename event and a create event, so treat rename as a remove.
```

Comment:
it's not, leftover comment.


On Fri, Apr 1, 2016 at 8:56 PM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/rsync/rsync.go
> <https://github.com/openshift/origin/pull/8268#discussion_r58285214>:
>
> > +
> > +	watcher, err := fsnotify.NewWatcher()
> > +	if err != nil {
> > +		return fmt.Errorf("error setting up filesystem watcher: %v", err)
> > +	}
> > +	defer watcher.Close()
> > +
> > +	go func() {
> > +		for {
> > +			select {
> > +			case event := <-watcher.Events:
> > +				changeLock.Lock()
> > +				glog.V(5).Infof("filesystem watch event: %s", event)
> > +				lastChange = time.Now()
> > +				dirty = true
> > +				// Rename sends a rename event and a create event, so treat rename as a remove.
>
> where is the rename getting treated as a remove?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8268/files/36f7bfc6d9a426395d3b012b2078710ed91e1ac4#r58285214>
>



-- 
Ben Parees | OpenShift